lib/admin_constraint.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require_dependency 'current_user'
+
+class AdminConstraint
+  def initialize(options = {})
+    @require_master = options[:require_master]
+  end
+
+  def matches?(request)
+    provider = Fundraising.session_provider.new(request.env)
```

Comment:
Let's leave the admin constraint out of this PR and implement it correctly in another one.